src/services/latex.js

```diff
@@ -97,19 +97,73 @@ var latexMathParser = (function() {
   latexMath.latexText = mathMode.or(textChar).many();
 
   latexMath.parse =
-  latexMath.latexText.parse = Parser.p._parse;
+  latexMath.latexText.parse = function(latex, latexCmds) {
+    var parser = this;
+    return localLatexCmds.let(latexCmds, function() {
+      return parser._parse(latex);
+    });
+  };
+  var localLatexCmds = ENV_VAR(); // LaTeX commands during this parse
 
   return latexMath;
 })();
 
+/**
+ * Concise JSON Schema for `customSymbols` declarations:
+ *   {
+ *     "/[a-z]+/i": {
+ *       spacing: { $enum: ['ord', 'bin', 'rel'] },
+ *       $optional_italic: { $value: true } // only allowed if spacing === 'ord'
+ *     }
+ *   }
+ * (More about Concise JSON Schemas: https://git.io/vwI9k )
+ *
+ * This is based on two classifications TeX uses for typesetting:
+ *   - the type of atom ("ord, op, bin, rel, open, close, punct and inner")
+ *       + determines spacing and line-breaking
+ *   - the family of font (normal upright roman, italic, etc)
+ *
+ * https://github.com/Khan/KaTeX/wiki/Examining-TeX#group-types
+ * http://tex.stackexchange.com/a/38984
+ */
```

Comment:
MathML is, unsurprisingly, unhelpful here. Presentation MathML has separate `<mi>`, `<mn>`, and `<mo>` elements ("identifiers", "numerals", and "operators", resp.), but they're for semantics (which really should be the purview of Content MathML...), not, y'know, "Presentation" like they should be. For font family stuff, all 3 elements have a `mathvariant` attribute; for spacing, `<mo>` has `form`, `lspace`, and `rspace` attributes, which are in theory supposed to have smart defaults, but for example the `form` attribute's smart default doesn't handle unary negation in the case of `1+-2`: http://jsbin.com/cilicu/edit?html,output